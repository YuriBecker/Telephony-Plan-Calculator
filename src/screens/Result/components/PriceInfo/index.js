import React from 'react';
import { View, ViewPropTypes } from 'react-native';
import PropTypes from 'prop-types';
import TextTitle from '../TextTitle';
import colors from '../../../../styles/colors';

const index = ({ text, price, negative, containerStyle }) => {
  const newStyle = negative ? colors.red : colors.primary;
  return (
    <View
      style={[
        {
          flexDirection: 'row',
          borderBottomColor: newStyle,
          borderRadius: 8,
          borderBottomWidth: 3,
          padding: 5,
        },
        containerStyle,
      ]}
    >
      <TextTitle style={{ color: newStyle }} category="h4">
        {text}
      </TextTitle>
      <TextTitle
        style={{ marginLeft: 20, color: newStyle, fontWeight: 'bold' }}
        category="h4"
      >
        {`R$ ${price.replace('.', ',')}`}
      </TextTitle>
    </View>
  );
};

export default index;

index.propTypes = {
  text: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  negative: PropTypes.bool,
  containerStyle: ViewPropTypes.style,
};

index.defaultProps = {
  negative: false,
  containerStyle: {},
};
