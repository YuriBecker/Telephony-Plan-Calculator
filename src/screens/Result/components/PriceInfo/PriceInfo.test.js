import React from 'react';
import { shallow } from 'enzyme';

import PriceInfo from './index';

describe('Testing PriceInfo component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<PriceInfo text="Some text" price={77} />);
  });

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());
});
