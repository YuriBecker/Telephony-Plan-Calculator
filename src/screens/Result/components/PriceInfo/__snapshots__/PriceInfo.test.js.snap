// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PriceInfo component should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomColor": "#1db954",
        "borderBottomWidth": 3,
        "borderRadius": 8,
        "flexDirection": "row",
        "padding": 5,
      },
      Object {},
    ]
  }
>
  <index
    category="h3"
    style={
      Object {
        "color": "#1db954",
      }
    }
  >
    Some text
  </index>
  <index
    category="h3"
    style={
      Object {
        "color": "#1db954",
        "fontWeight": "bold",
        "marginLeft": 20,
      }
    }
  >
    R$ 77
  </index>
</View>
`;
